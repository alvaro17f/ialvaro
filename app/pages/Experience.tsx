"use client";
import { useRef } from "react";
import { domAnimation, LazyMotion, m, useScroll } from "framer-motion";
import Link from "next/link";
import data from "@/data/portfolio.json";
import Header from "@/components/Header";

export default function Experience() {
	const ref = useRef(null);
	const { scrollYProgress } = useScroll({
		target: ref,
		offset: ["end end", "start start"],
	});

	return (
		<section id="experience">
			<Header title="Experience" />
			<div className="grid md:grid-cols-2 place-items-center">
				{data.map(({ id, title, image, url, description }) => (
					<LazyMotion key={id} features={domAnimation}>
						<div ref={ref}>
								<Link href={url} target="_blank">
									<m.div
										className="cursor-pointer w-[30dvw]"
										initial={{ scale: 1, opacity: 0 }}
										whileInView={{opacity: 1}}
										viewport={{once: true}}
										whileHover={{ scale: 1.1 }}
										whileTap={{ scale: 1 }}
										transition={{ duration: 0.7 }}
									>
										<div className="h-32" />
										<m.h1
											key={title}
											className="mb-5 text-4xl md:text-6xl text-azama-white w-[100%]"
										>
											{title}
										</m.h1>
										<img src={image} alt={title} />
										<p>{description}</p>
									</m.div>
								</Link>
						</div>
					</LazyMotion>
				))}
			</div>
		</section>
	);
}
