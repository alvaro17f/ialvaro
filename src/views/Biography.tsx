import { Content } from "src/components/Content";
import data from "src/data/biography.json";
import { type ChangeEvent, useEffect, useState } from "react";

export default function Biography() {
	const [isMobile, setIsMobile] = useState(true);
	const [slider, setSlider] = useState(Math.round((data.length - 1) / 2));
	const [selectedBio, setSelectedBio] = useState(data[slider]);

	useEffect(() => {
		setIsMobile(window.innerWidth < 768);
	});

	useEffect(() => {
		if (slider < data.length) {
			setSelectedBio(data[slider]);
		}
	}, [slider]);

	const handleSlider = (e: ChangeEvent<HTMLInputElement>) => {
		setSlider(Number(e.target.value));
	};

	return (
		<section
			id="biography"
			className="md:h-[100dvh] grid items-center pt-24 md:pt-0"
		>
			<Content>
				<div className="grid grid-cols-1 md:gap-10 md:grid-cols-[1fr_2fr]">
					<div className="mb-5">
						<a
							href="https://linkedin.com/in/alvarogarciamacias"
							rel="noopener noreferrer"
							target="_blank"
						>
							<img
								src="/images/profile/profile.png"
								alt="profile"
								width={250}
								height={250}
								className="border-[7px] rounded-full border-alvaro-white hover:border-dashed cursor-pointer"
							/>
						</a>
					</div>
					<div className="max-w-[75ch] md:h-[300px] items-center md:text-justify grid  whitespace-pre-wrap">
						<div>
							<h1 className="mb-5 text-4xl">
								Hello <span className="text-alvaro-danger">World</span>!
							</h1>
							<p>
								My name is{" "}
								<span className="text-xl text-alvaro-primary">√Ålvaro</span>, and
								I am a Full Stack Developer.
							</p>
							{selectedBio.bio}
						</div>
					</div>
				</div>
				<div className="grid w-full grid-cols-2 mt-20 md:grid-cols-3 place-items-center">
					<button
						type="button"
						onClick={() => slider !== 0 && setSlider((slider) => slider - 1)}
						className="p-2 border text-alvaro-primary rounded-xl border-alvaro-primary md:hover:bg-alvaro-primary md:hover:text-alvaro-base"
					>
						Shortest
					</button>
					{!isMobile && (
						<input
							id="minmax-range"
							type="range"
							min="0"
							max={data.length - 1}
							value={slider}
							onChange={handleSlider}
							className="w-full h-2 bg-gray-200 rounded-lg outline-none appearance-none cursor-pointer dark:bg-gray-700"
						/>
					)}
					<button
						type="button"
						onClick={() =>
							slider !== data.length - 1 && setSlider((slider) => slider + 1)
						}
						className="p-2 border text-alvaro-danger rounded-xl border-alvaro-danger md:hover:bg-alvaro-danger md:hover:text-alvaro-base"
					>
						Longest
					</button>
				</div>
			</Content>
		</section>
	);
}
