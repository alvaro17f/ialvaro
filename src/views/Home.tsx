import Wobble from "src/components/Wobble";

export default function Home() {
	return (
		<section
			id="home"
			className="h-[100dvh] top-0 grid place-items-center gap-5"
		>
			<div>
				<Wobble sentence="ALVARO" style="hover:text-alvaro-primary" />
				<Wobble sentence="GARCIA" style="hover:text-alvaro-danger" />
				<Wobble sentence="MACIAS" style="hover:text-alvaro-primary" />
			</div>
		</section>
	);
}
