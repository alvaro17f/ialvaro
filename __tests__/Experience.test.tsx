import { Experience } from "src/views/Experience";
import { cleanup, render, screen } from "@testing-library/react";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";

describe("<Experience />", () => {
	beforeEach(() => {
		const mockIntersectionObserver = vi.fn();
		mockIntersectionObserver.mockReturnValue({
			observe: () => null,
			unobserve: () => null,
			disconnect: () => null,
		});
		window.IntersectionObserver = mockIntersectionObserver;
		render(<Experience />);
	});
	afterEach(cleanup);

	it("should match the snapshot", () => {
		const { container } = render(<Experience />);
		expect(container).toMatchSnapshot();
	});

	it("should render a heading", () => {
		expect(
			screen.getByRole("heading", { level: 1, name: /experience/i }),
		).toBeDefined();
	});

	it("should render skills", () => {
		expect(
			screen.getAllByLabelText(/experience/i).length,
		).toBeGreaterThanOrEqual(1);
		expect(screen.getAllByRole("img").length).toBeGreaterThanOrEqual(1);
	});

	it("should render a experience's title", () => {
		expect(screen.getAllByLabelText(/title/i).length).toBeGreaterThanOrEqual(1);
	});

	it("should render a experience's content", () => {
		expect(screen.getAllByLabelText(/content/i).length).toBeGreaterThanOrEqual(
			1,
		);
	});
});
