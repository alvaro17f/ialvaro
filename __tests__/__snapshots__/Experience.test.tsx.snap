// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Experience /> > should match the snapshot 1`] = `
<div>
  <section
    id="experience"
  >
    <div
      class="h-24"
    />
    <section
      class="p-2 mb-5 rounded-lg text-alvaro-dark from-alvaro-primary to-transparent bg-gradient-to-r"
      style="width: 0%; opacity: 0; transform: scaleY(0.5) translateZ(0);"
    >
      <h1
        class="text-5xl md:text-3xl"
        style="opacity: 0;"
      >
        Experience
      </h1>
    </section>
    <div
      class="grid md:grid-cols-[1fr_2fr] mt-12 md:gap-10"
    >
      <div
        aria-label="experience"
        class="grid gap-2 border-dashed cursor-pointer border-[4px] border-alvaro-muted b-20 md:p-5 h-60 md:mb-36 place-items-center rounded-xl"
        style="opacity: 0;"
        tabindex="0"
      >
        <a
          href="https://devoteam.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="devoteam"
            class="w-56 h-36"
            src="/experience/devoteam.svg"
          />
        </a>
        <p>
          October 2021
           - 
          April 2023
        </p>
      </div>
      <div
        class="mt-12 md:mt-0"
      >
        <h1
          aria-label="title"
          class="mb-5 text-4xl md:text-6xl text-alvaro-primary"
          style="opacity: 0;"
        >
          devoteam
        </h1>
        <div
          aria-label="content"
          class="mb-20 whitespace-pre-wrap"
          style="opacity: 0;"
        >
          <p
            class="text-xl underline text-alvaro-danger"
          >
            Client
          </p>
          RSI
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          >
            Projects
          </p>
          Frontend management for Banco Caminos, Arquia, Vall Banc, Fiare, Bancofar, etc...
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          >
            Tasks
          </p>
          - Web application development using Vue.js and React.js
- REST API integration in front-end applications
- Unit and integration test development
- Code documentation
- Translations using i18n
- Modification and elaboration of login pages
- Internal SSO access elaboration
- Modification of CSS styles
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          >
            Skills
          </p>
          vue.js
react.js
typescript
javascript
jest.js
node.js
html
css
jsp
oracle db
inkscape
jira
git
dbeaver
postman
invision
overflow
        </div>
      </div>
    </div>
    <div
      class="grid md:grid-cols-[1fr_2fr] mt-12 md:gap-10"
    >
      <div
        aria-label="experience"
        class="grid gap-2 border-dashed cursor-pointer border-[4px] border-alvaro-muted b-20 md:p-5 h-60 md:mb-36 place-items-center rounded-xl"
        style="opacity: 0;"
        tabindex="0"
      >
        <a
          href="https://knowmadmood.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Knowmad Mood"
            class="w-56 h-36"
            src="/experience/knowmadmood.svg"
          />
        </a>
        <p>
          June 2023
           - 
          now
        </p>
      </div>
      <div
        class="mt-12 md:mt-0"
      >
        <h1
          aria-label="title"
          class="mb-5 text-4xl md:text-6xl text-alvaro-primary"
          style="opacity: 0;"
        >
          Knowmad Mood
        </h1>
        <div
          aria-label="content"
          class="mb-20 whitespace-pre-wrap"
          style="opacity: 0;"
        >
          <p
            class="text-xl underline text-alvaro-danger"
          >
            Client
          </p>
          Open Bank
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          />
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          />
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          />
        </div>
      </div>
    </div>
    <div
      class="grid md:grid-cols-[1fr_2fr] mt-12 md:gap-10"
    >
      <div
        aria-label="experience"
        class="grid gap-2 border-dashed cursor-pointer border-[4px] border-alvaro-muted b-20 md:p-5 h-60 md:mb-36 place-items-center rounded-xl"
        style="opacity: 0;"
        tabindex="0"
      >
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="your company"
            class="w-56 h-36"
            src="/experience/company.svg"
          />
        </a>
        <p>
          April 2023
           - 
          now
        </p>
      </div>
      <div
        class="mt-12 md:mt-0"
      >
        <h1
          aria-label="title"
          class="mb-5 text-4xl md:text-6xl text-alvaro-primary"
          style="opacity: 0;"
        >
          your company
        </h1>
        <div
          aria-label="content"
          class="mb-20 whitespace-pre-wrap"
          style="opacity: 0;"
        >
          <p
            class="text-xl underline text-alvaro-danger"
          />
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          />
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          />
          <br />
          <br />
          <p
            class="text-xl underline text-alvaro-danger"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
